File: OCaml/sleep_sort.ml
let sort_me x =
  Thread.delay (Float.of_int x /. 10.0);
  print_int x;
  print_string "\n";;

let data = [10; 4; 2; 6; 5; 7; 1; 3];;

List.map (Thread.create sort_me) data;;

Thread.delay 2.

File: Rust/sleep-sort.rs
use std::{thread, time};

fn main() {
    let data = vec![10, 4, 2, 6, 5, 7, 1, 3];
    let mut threads = vec![];

    for n in data {
        threads.push(thread::spawn(move || {
            let delay = time::Duration::from_millis(n * 1);
            thread::sleep(delay);
            println!("{}", n);
        }));
    }

    for child in threads {
        let _ = child.join();
    }
}
